content/blogs/2025-03-27-using-amazon-s3-tables-with-kestra.md

```diff
@@ -0,0 +1,307 @@
+
+From the `General purpose bukcets` page, search for the newly created bucket, and click on the bucket name. On the corresponding bucket's home page, click on the `Create folder` button. Provide the folder name, say `data`, and click on the `Create folder` button at the bottom of the page. We will be storing our data in this folder.
+
+### Getting the data into S3 bucket
+
+Navigate to the Kestra UI, and create a new flow. We will download the CSV file containing products data using the [HTTP Download task](/plugins/core/http/io.kestra.plugin.core.http.download). We will then convert the CSV data into ION format using the [CsvToIon task](/plugins/plugin-serdes/csv/io.kestra.plugin.serdes.csv.csvtoion), and then from ION format into a parquet file using [IonToParquet task](/plugins/plugin-serdes/parquet/io.kestra.plugin.serdes.parquet.iontoparquet). Finally, we will upload the parquet file into the recently created S3 general purpose bucket inside the `data` folder using [S3 Upload task](/plugins/plugin-aws/s3/io.kestra.plugin.aws.s3.upload).
+
+This is how the Kestra flow will look like:
+
+```
+id: s3_tables_demo
+namespace: company.team
+description: With this flow, you will upload the products data in parquet format into S3 general purpose bucket
+
+tasks:
+  - id: http_download
+    type: io.kestra.plugin.core.http.Download
+    uri: https://huggingface.co/datasets/kestra/datasets/raw/main/csv/products.csv
+
+  - id: csv_to_ion
+    type: io.kestra.plugin.serdes.csv.CsvToIon
+    from: "{{ outputs.http_download.uri }}"
+
+  - id: ion_to_parquet
+    type: io.kestra.plugin.serdes.parquet.IonToParquet
+    from: "{{ outputs.csv_to_ion.uri }}"
+    schema: |
+      {
+        "type": "record",
+        "name": "Product",
+        "namespace": "com.kestra.product",
+        "fields": [
+          {"name": "product_id", "type": "int"},
+          {"name": "product_name", "type": "string"},
+          {"name": "product_category", "type": "string"},
+          {"name": "brand", "type": "string"}
+        ]
+      }
+
+  - id: s3_upload
+    type: io.kestra.plugin.aws.s3.Upload
+    accessKeyId: "{{ secret('AWS_ACCESS_KEY') }}"
+    secretKeyId: "{{ secret('AWS_SECRET_KEY') }}"
+    region: "{{ secret('AWS_REGION') }}"
+    from: "{{ outputs.ion_to_parquet.uri }}"
+    bucket: "s3-general-purpose-ecommerce"
+    key: "data/products.parquet"
+```
+
+### Creating S3 Table Bucket
+
+On the [S3 service](https://console.aws.amazon.com/s3/home) page on the AWS console, navigate to `Table buckets` from the left navigation menu.
+
+In case, you are opening the table buckets page for the first time, you will see a box at the top of the page about `Integration with AWS analytics services - New` with the `Enable integration` button. Click on the button and enable the integration of S3 table buckets with AWS analytics services like Amazon EMR, Amazon Redshift and Amazon Athena.
```

Comment:
```suggestion
In case, you are opening the table buckets page for the first time, you will see a box at the top of the page about `Integration with AWS analytics services - New` with the `Enable integration` button. Select the button and enable the integration of S3 table buckets with AWS analytics services like Amazon EMR, Amazon Redshift and Amazon Athena.
```